### PR TITLE
Recent capture files pr

### PR DESCRIPTION
**Pull request by antonrohr**
I tried to structure the commits nicely, so not a lot of explanation is necessary. 

This is mainly adding the ItemModel for CaptureFileInfo and adds FillFromDirectory to the Manager. The later will be used to only once to fill the recent capture files from the output directory

### CODE REVIEW

File: src/CaptureFileInfo/CaptureFileInfoTest.cpp [hunk 1]
```diff
@@ -85,4 +85,23 @@ TEST(CaptureFileInfo, Touch) {
...
+  {
+    const QString non_existing_path{"test/path/file.ext"};
+
+    CaptureFileInfo capture_file_info{non_existing_path};
+
+    EXPECT_EQ(capture_file_info.FileSize(), 0);
+  }
+  {
+    const std::filesystem::path path =
+        orbit_base::GetExecutableDir() / "testdata" / "not_existing_test_file.txt";
```
Comment (beckerhe) [inline src/CaptureFileInfo/CaptureFileInfoTest.cpp]: `not_existing_test_file`? Does that file actually exist?

File: src/CaptureFileInfo/ItemModel.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+#include "CaptureFileInfo/CaptureFileInfo.h"
+#include "CoreUtils.h"
+#include "OrbitBase/Logging.h"
+
+namespace orbit_capture_file_info {
+
+void ItemModel::SetCaptureFileInfos(std::vector<CaptureFileInfo> capture_file_infos) {
+  beginInsertRows(QModelIndex{}, 0, static_cast<int>(capture_file_infos.size()) - 1);
+  capture_files_ = std::move(capture_file_infos);
+  endInsertRows();
```
Comment (beckerhe) [inline src/CaptureFileInfo/ItemModel.cpp:16]: You're replacing all entries in capture_files_ which means we have to express that as deleting all rows first and then adding new ones. So this needs some `beginRemoveRows` as well.

Comment (beckerhe) [reply src/CaptureFileInfo/ItemModel.cpp:16]: This is also not taking into account that `capture_file_infos` could be empty.

File: src/CaptureFileInfo/Manager.cpp [hunk 3]
```diff
@@ -48,14 +48,14 @@ void Manager::SaveCaptureFileInfos() {
   settings.endArray();
 }
 
-void Manager::AddOrTouchCaptureFile(const QString& path) {
+void Manager::AddOrTouchCaptureFile(const std::filesystem::path& path) {
```
Comment (beckerhe) [inline src/CaptureFileInfo/Manager.cpp:53]: I don't see why you're doing this. Maybe you could add an explanation to the commit message?

File: src/CaptureFileInfo/Manager.cpp [hunk 4]
```diff
@@ -77,4 +79,19 @@ void Manager::PurgeNonExistingFiles() {
...
 }
 
+ErrorMessageOr<void> Manager::FillFromDirectory(const std::filesystem::path& directory) {
+  OUTCOME_TRY(files, orbit_base::ListFilesInDirectory(directory));
+
+  for (const auto& file : files) {
+    if (file.extension() != ".orbit") continue;
+
+    QFileInfo tmp_file_info{QString::fromStdString(file.string())};
+    capture_file_infos_.emplace_back(tmp_file_info.filePath(), tmp_file_info.birthTime());
```
Comment (beckerhe) [inline src/CaptureFileInfo/Manager.cpp:89]: What happens when FillFromDirectory has already been called, or a particular file is already present in `capture_file_infos_`. There is no deduplication, right? Maybe this method should clear `capture_file_infos_` before filling it.

File: src/CaptureFileInfo/ManagerTest.cpp [hunk 5]
```diff
@@ -144,4 +147,33 @@ TEST(CaptureFileInfoManager, Persistency) {
...
+TEST(CaptureFileInfoManager, FillFromDirectory) {
+  QCoreApplication::setOrganizationName(kOrgName);
+  QCoreApplication::setApplicationName("CaptureFileInfo.Manager.FillFromDirectory");
+
+  Manager manager;
+  manager.Clear();
+
+  {  // Fail
+    ErrorMessageOr<void> result = manager.FillFromDirectory("/non/existent/path/to/dir");
+    ASSERT_TRUE(result.has_error());
```
Comment (beckerhe) [inline src/CaptureFileInfo/ManagerTest.cpp:159]: I think this is superfluous when you use the HasError matcher

File: src/CaptureFileInfo/CMakeLists.txt [hunk 6]
```diff
@@ -48,8 +48,11 @@ target_link_libraries(
...
 
-add_custom_command(TARGET CaptureFileInfoTests POST_BUILD
-  COMMAND ${CMAKE_COMMAND} -E copy_directory ${CMAKE_CURRENT_LIST_DIR}/testdata
-  $<TARGET_FILE_DIR:CaptureFileInfoTests>/testdata)
+add_custom_command(TARGET CaptureFileInfoTests POST_BUILD COMMAND ${CMAKE_COMMAND}
+          -E remove_directory $<TARGET_FILE_DIR:CaptureFileInfoTests>/testdata/CaptureFileInfo)
+
+add_custom_command(TARGET CaptureFileInfoTests POST_BUILD COMMAND ${CMAKE_COMMAND}
+          -E copy_directory ${CMAKE_CURRENT_SOURCE_DIR}/testdata/
+          $<TARGET_FILE_DIR:CaptureFileInfoTests>/testdata/CaptureFileInfo)
```
Comment (beckerhe) [inline src/CaptureFileInfo/CMakeLists.txt]: When using two rules you don't have control over the execution order. They could be executed in parallel.

But you can use multiple COMMAND directives in the same `add_custom_command` call. They will be executed in order.

File: src/CaptureFileInfo/CaptureFileInfoTest.cpp (hunk 1 above)
Comment (antonrohr) [reply src/CaptureFileInfo/CaptureFileInfoTest.cpp]: I rebased wrong. This was a mistake that I fixed, but the fix ended up one commit later. This means also the test will fail at this commit. I will fix that

Comment (beckerhe) [reply src/CaptureFileInfo/CaptureFileInfoTest.cpp]: Feel free to force push.

File: src/CaptureFileInfo/CMakeLists.txt (hunk 6 above)
Comment (antonrohr) [reply src/CaptureFileInfo/CMakeLists.txt]: I copied this from the CMakeLists.txt of OrbitBase. I fixed it here and added a separate commit for fixing it in OrbitBase

File: src/CaptureFileInfo/Manager.cpp (hunk 4 above)
Comment (antonrohr) [reply src/CaptureFileInfo/Manager.cpp:91]: done

File: src/CaptureFileInfo/Manager.cpp (hunk 3 above)
Comment (antonrohr) [reply src/CaptureFileInfo/Manager.cpp:53]: I will add an explanation to the commit message when doing the rebase.

File: src/CaptureFileInfo/ItemModel.cpp (hunk 2 above)
Comment (antonrohr) [reply src/CaptureFileInfo/ItemModel.cpp]: done

File: src/CaptureFileInfo/CMakeLists.txt (hunk 6 above)
Comment (beckerhe) [reply src/CaptureFileInfo/CMakeLists.txt]: Even better. Thanks!

**Review by beckerhe — APPROVED** [review]
Thank you for all the fixes. LGTM!

**Review by dimitry- — APPROVED** [review]

**Pull request merged**